docs/how-to/rest-api-and-dynamodb.md

```diff
@@ -0,0 +1,1364 @@
+
+Cyclic does this automatically for us in production-mode, however. So we don’t need to worry about this task when deploying our API.
+
+Remember when we mentioned Node’s package manager (NPM) that we used to get access to Express? Well, GitHub repositories that use NPM only contain the *names* and *versions* of used packages, but not the actual packages themselves.
+
+Whenever we clone a GitHub repository, we must run `npm install` to actually *download* those packages into our machine. 
+
+## Our database is empty… let’s create some mock data
+
+![BikesDB.png](../../static/img/tutorial/rest-api/BikesDB.png)
+
+Everything’s going great, it’s time to get started building our database!
+
+First of all, every AWS DynamoDB instance has a specific name, also known as a *table name*. It’s given to us by Cyclic in its database dashboard page.
+
+![Copying table name to the clipboard.](../../static/img/tutorial/rest-api/screencast2.gif)
+
+Let’s copy that value into the `.env.sample` file and remove the `BUCKET` variable. Such a file contains environment variables that must be registered before running our server. It won’t work until we copy it into a `.env` file, however:
+
+[Learn more about environment variables.](https://www.freecodecamp.org/news/what-are-environment-variables-and-how-can-i-use-them-with-gatsby-and-netlify/)
+
+```bash
+cp env.sample .env
+```
+
+> The database credentials that we copied into our machine are themselves environment variables. However, we didn’t copy them into our `.env` file since they’re only temporary for our local development, and Cyclic will change them once deployed on the Internet.
+The table name, however, is static and won’t change; it’s safe to put it in the `.env` file.
+> 
+
+After doing that, we can start our REST API by running `npm run dev`.
+
+Let’s make sure that our *local* server is up and running by debugging it using cURL:
+
+```bash
+curl http://localhost:3000/animals/rin%20tin%20tin \
+    --data '{"breed":"German Shepard", "gender": "male"}' \
+    -XPOST -H 'Content-Type: application/json'| jq .
+```
+
+As you can see, *local* servers are hosted on the [`localhost`](http://localhost) domain or [`127.0.0.1`](http://127.0.0.1) IP address. It’s also common to see them running on the `3000` port, as being done in this server. (`8000`, `8080`, `5000` are also equally as common) 
+
+[Learn more about local hosting and ports.](https://developer.mozilla.org/en-US/docs/Learn/Common_questions/set_up_a_local_testing_server)
+
+---
+
+Our database is currently empty, and it’s not very convenient to create API routes that fetch data when… well, there’s no data to fetch! That would make it nearly impossible to debug our RESTful API.
+
+Therefore, for the purposes of this tutorial, you can run the following script that generates some mock data to fill your database. Create a new file called `fill-db.js` and copy the contents of this script into it:
+
+```javascript
+import DynamoDb from "cyclic-dynamodb";
```

Comment:
Can we add this script to the `bikes-api` repo? and add a package.json script to run it?